packages/frontend/src/components/common/ShareSpaceModal/ShareSpaceUserList.tsx

```diff
@@ -296,15 +333,43 @@ type GroupAccessListProps = {
         action: UserAccessAction,
         currentGroupAccess: SpaceGroup,
     ) => void;
+    pageSize?: number;
 };
 const GroupsAccessList: FC<GroupAccessListProps> = ({
     isPrivate,
     onAccessChange,
     groupsAccess,
+    pageSize,
 }) => {
+    const [page, setPage] = useState(1);
+
+    // TODO: Paginate group access from backend
```

Comment:
do we need to do this in the same PR?

Reply:
We need to paginate these because otherwise we will try to render all of the users/groups in the lists, the problem with these is that they come from nested properties in the request responses, so we can't really paginate these requests (there should be endpoints specific to fetch these 2 lists)

We have created tickets which are part of this milestone to do this, but for now a simple frontend pagination helps to render the modal and avoid the sluggish behavior